helper_dart/lib/src/insights.dart

```diff
@@ -0,0 +1,99 @@
+import 'package:algolia/algolia.dart';
+import 'package:collection/collection.dart';
+
+import 'algolia_event_service_adapter.dart';
+import 'event_service.dart';
+import 'event_tracker.dart';
+
+/// Insights is the component responsible for sending events related to
+/// the user to personalize his search
+class Insights implements EventTracker {
+  /// Index name
+  String indexName;
+
+  /// A pseudonymous or anonymous user identifier.
+  String userToken;
+
+  @override
+  bool isEnabled;
+
+  /// Service sending event packets
+  EventService service;
+
+  static const _maxObjectIDsPerEvent = 20;
+  static const _maxFiltersPerEvent = 10;
+
+  Insights(String applicationID, String apiKey, String indexName)
+      : this.custom(
+          AlgoliaEventServiceAdapter(applicationID, apiKey),
+          indexName,
+        );
+
+  Insights.custom(this.service, this.indexName)
+      : userToken = _generateUserToken(),
+        isEnabled = true;
+
+  static String _generateUserToken() => 'userToken';
```

Comment:
```suggestion
  // TODO: user token generation implementation
  static String _generateUserToken() => 'userToken';
```

Maybe a small comment here to indicate that this is just a holder for a later implementation